runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -8269,6 +8269,29 @@ done:;
 			/* Get MethodHandle for this operation from the VarHandles handleTable */
 			j9object_t handleTable = J9VMJAVALANGINVOKEVARHANDLE_HANDLETABLE(_currentThread, varHandle);
 			j9object_t methodHandle = J9JAVAARRAYOFOBJECT_LOAD(_currentThread, handleTable, operation);
+
+			if (NULL == methodHandle) {
+				prepareForExceptionThrow(_currentThread);
+
+				const char *exceptionClassName = "java/lang/UnsupportedOperationException";
+				J9Class *exceptionClass = internalFindClassUTF8(
+						_currentThread,
+						(U_8 *)exceptionClassName,
+						strlen(exceptionClassName),
+						_currentThread->javaVM->systemClassLoader,
+						J9_FINDCLASS_FLAG_THROW_ON_FAIL);
+
+				j9object_t exception = _objectAllocate.inlineAllocateObject(_currentThread, exceptionClass);
+				if (NULL == exception) {
```

Comment:
A null return here says that we can't allocate without doing a GC.  It's not going to have an exception pending.  Since we're on an error path, better to use the out of line allocate `_vm->memoryManagerFunctions->J9AllocateObject(`.

The `prepareForExceptionThrow` will have built a special frame for us already so we don't have to build another one here.

We will need to add a `VMStructHasBeenUpdated(REGISTER_ARGS);` after the `J9AllocateObject` call